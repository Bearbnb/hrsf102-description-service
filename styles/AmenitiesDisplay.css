.container {
  max-width: 600px;
  margin: 30px auto 15px auto;
  font-size: 16px;
}

.columns {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.rightColumn {
  padding-right: 100px;
}

.amenity {
  width: 100%;
  margin-top: 15px;
}

.title {
  font-weight: 500;
  font-size: 16px;
}

.pageBreak {
  color: #808080;
}

.button {
  background: none!important;
  margin-top: 25px;
  color: #6BB7B9;
  border: none; 
  padding: 0!important;
  font: inherit;
  cursor: pointer;
}