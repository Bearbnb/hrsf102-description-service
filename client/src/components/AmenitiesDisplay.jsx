import React from 'react';
import PropTypes from 'prop-types';

import Amenities from './Amenities.jsx';
import styles from '../../../styles/AmenitiesDisplay.css';
import { countAmenities } from '../../../seedDatabase/helpers';

const AmenitiesDisplay = ({
  displayAmenities,
  amenities,
  showModal,
  showAmenities,
  hideAmenities,
}) => {
  const column1 = displayAmenities.slice(0, 3);
  const column2 = displayAmenities.slice(3, 6);

  const amenitiesCount = countAmenities(amenities);

  return (
    <div className={styles.container}>
      <hr className={styles.pageBreak} />
      <div className={styles.title}>
        Amenities
      </div>
      <div className={styles.columns}>

        <div>
          {column1.map(amenity => (
            <div key={amenity.id} className={styles.amenity}>{amenity}</div>
          ))}
        </div>

        <div className={styles.rightColumn}>
          {column2.map(amenity => (
            <div key={amenity.id} className={styles.amenity}>{amenity}</div>
          ))}
        </div>

      </div>

      <button
        className={styles.button}
        type="button"
        onClick={() => showAmenities()}
      >Show all {amenitiesCount} amenities
      </button>

      <Amenities
        amenities={amenities}
        hideAmenities={hideAmenities}
        showModal={showModal}
      />

    </div>
  );
};

AmenitiesDisplay.propTypes = {
  displayAmenities: PropTypes.arrayOf(PropTypes.string),
  showModal: PropTypes.bool,
  showAmenities: PropTypes.func,
  hideAmenities: PropTypes.func,
  amenities: PropTypes.shape({
    id: PropTypes.number,
    _id: PropTypes.string,
    basics: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        details: PropTypes.string,
      }),
    ),
    bedAndBath: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        details: PropTypes.string,
      }),
    ),
    dining: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        details: PropTypes.string,
      }),
    ),
    guestAccess: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        details: PropTypes.string,
      }),
    ),
    safetyFeautes: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        details: PropTypes.string,
      }),
    ),
    notIncluded: PropTypes.arrayOf(PropTypes.string),
  }),
};

AmenitiesDisplay.defaultProps = {
  displayAmenities: [],
  showModal: false,
  showAmenities: () => {},
  hideAmenities: () => {},
  amenities: {},
};

export default AmenitiesDisplay;
