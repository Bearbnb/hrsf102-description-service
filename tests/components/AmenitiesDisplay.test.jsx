import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import AmenitiesDisplay from '../../client/src/components/AmenitiesDisplay';

describe('<AmenitiesDisplay />', () => {
  it('should render properly', () => {
    const wrapper = shallow(<AmenitiesDisplay amenities={{}} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
