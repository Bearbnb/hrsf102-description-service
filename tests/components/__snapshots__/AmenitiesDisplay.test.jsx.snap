// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AmenitiesDisplay /> should render properly 1`] = `
<div
  className="container"
>
  <hr
    className="pageBreak"
  />
  <div
    className="title"
  >
    Amenities
  </div>
  <div
    className="columns"
  >
    <div />
    <div
      className="rightColumn"
    />
  </div>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    Show all 
    0
     amenities
  </button>
  <Amenities
    amenities={Object {}}
    hideAmenities={[Function]}
    showModal={false}
  />
</div>
`;
